src/partition/repart.c

```diff
@@ -3777,15 +3777,13 @@ static int partition_encrypt(Context *context, Partition *p, PartitionTarget *ta
                 ssize_t base64_encoded_size;
                 int keyslot;
 
-                if (arg_tpm2_public_key_pcr_mask != 0) {
-                        r = tpm2_load_pcr_public_key(arg_tpm2_public_key, &pubkey, &pubkey_size);
-                        if (r < 0) {
-                                if (arg_tpm2_public_key || r != -ENOENT)
-                                        return log_error_errno(r, "Failed read TPM PCR public key: %m");
+                r = tpm2_load_pcr_public_key(arg_tpm2_public_key, &pubkey, &pubkey_size);
+                if (r < 0) {
+                        if (arg_tpm2_public_key || r != -ENOENT)
+                                return log_error_errno(r, "Failed to read TPM PCR public key: %m");
 
-                                log_debug_errno(r, "Failed to read TPM2 PCR public key, proceeding without: %m");
-                                arg_tpm2_public_key_pcr_mask = 0;
-                        }
+                        log_debug_errno(r, "Failed to read TPM2 PCR public key, proceeding without: %m");
+                        arg_tpm2_public_key_pcr_mask = 0;
                 }
```

Comment:
why is this part of the change necessary? i mean, if arg_tpm2_public_key_pcr_mask is zero at this point, then there's no value in loading arg_tpm2_public_key, is there?

Reply:
right, fixed